test/e2e/masteruserrecord_test.go

```diff
@@ -0,0 +1,149 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/pkg/apis/toolchain/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	"github.com/codeready-toolchain/toolchain-e2e/tiers"
+	. "github.com/codeready-toolchain/toolchain-e2e/wait"
+	"github.com/operator-framework/operator-sdk/pkg/test"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUpdateMURToOtherRevisionOfTierTemplates(t *testing.T) {
+	// given
+	tierList := &toolchainv1alpha1.NSTemplateTierList{}
+	ctx, awaitility := testsupport.WaitForDeployments(t, tierList)
+	defer ctx.Cleanup()
+
+	// Create and approve "testing-template-updates" user
+	murName := "testingtemplateupdates"
+	userSignup := createAndApproveSignup(t, awaitility, murName)
+
+	tierTemplatesByTier := getTierTemplatesMappedByTier(t, awaitility)
+	templatesFromAdvanced := createBasicTierTemplate(t, ctx, awaitility, tierTemplatesByTier["advanced"], "123abcd")
+	templatesFromTeam := createBasicTierTemplate(t, ctx, awaitility, tierTemplatesByTier["team"], "456efgh")
+	templatesFromBasic := createBasicTierTemplate(t, ctx, awaitility, tierTemplatesByTier["basic"], "789ijkl")
+
+	// wait for the user to be provisioned for the first time
+	verifyResourcesProvisionedForSignup(t, awaitility, userSignup, "basic")
+
+	t.Run("update to refs basic-*-123abcd made out of the advanced TierTemplates", func(t *testing.T) {
+		// when update to basic-*-123abcd
+		replaceMurRefs(t, awaitility, murName, templatesFromAdvanced)
```

Comment:
Do we actually support changing template refs directly in MUR right now? Wouldn't TemplateTier reconcile sync the MURs to the refs used by the tier?

Reply:
I moved this test in a separate file, but it was authored by @MatousJobanek in https://github.com/codeready-toolchain/toolchain-e2e/pull/119/files#diff-26ea7544c30692a2091a1b3ec8907eb7R110. 

The context for his PR is: verifying that the MUR is provisioned when it is changed. And yes, the change will be done by the TemplateUpdateRequestController which is coming in a subsequent PR. In other words, we are testing portions of the workflow and there are still a few missing pieces in the puzzle ;)

Reply:
Ok. I’m just concerned about creating a flaky test when we “manually” change the template refs in MUR which could be overridden by a Tier controller reconcile.